### PR TITLE
fix crash in prepare_po_requisition

### PR DESCRIPTION
**Pull request by gurneyalex**

### CODE REVIEW

**Comment by lepistone** [comment]
thanks :+1:

File: logistic_requisition_department/model/logistic_requisition.py [hunk 1]
```diff
@@ -48,6 +49,12 @@ class LogisticRequisitionSource(models.Model):
...
-
-        res['department_id'] = self.department_id.id
+        for line in self:
+            if line.department_id:
+                if 'department_id' in res and res['department_id'] != line.department_id.id:
+                    raise exceptions.Warning(
+                        'Cannot generate Purchase Order: '
+                        'the sourcing lines are from '
+                        'different departments.')
+                res['department_id'] = line.department_id.id
```
Comment (yvaucher) [inline logistic_requisition_department/model/logistic_requisition.py]: Why not using `.mapped` ?

```
dep = self.mapped('department_id')
if len(dep) > 1:
    raise [...]
res['department_id'] = dep.id
```

Comment (yvaucher) [inline logistic_requisition_department/model/logistic_requisition.py]: https://www.odoo.com/documentation/8.0/reference/orm.html#other-recordset-operations

Mapped should remove duplicates if it is mapping record set

**Comment by yvaucher** [comment]
:+1:

**Comment by gurneyalex** [comment]
rebased w/ autosquash

**Pull request merged**